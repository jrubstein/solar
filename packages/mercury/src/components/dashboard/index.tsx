import React from 'react'
import { Helmet } from 'react-helmet'
import { withNamespaces, WithNamespaces } from 'react-i18next'
import { Dispatch } from 'redux'
import { WithStyles, StyleRules } from '@material-ui/core/styles'
import injectSheet from 'react-jss'
import { connect } from 'react-redux'
import { GET_LOGGED_USER } from '../../services/user/Actions'

const styles: StyleRules = {
  content: {
    display: 'flex',
    justifyContent: 'center',
  },
}

interface DashboardProps extends WithNamespaces, WithStyles {
  requestLoggedUser: () => void
}

class DashboardView extends React.Component<DashboardProps> {
  constructor(props: DashboardProps) {
    super(props)
  }

  render() {
    const { classes, t, requestLoggedUser } = this.props
    return (
      <React.Fragment>
        <Helmet>
          <title>{t('dashboard-title')}</title>
        </Helmet>
        <section className={classes.content}>
          <button onClick={requestLoggedUser}>{t('dashbord-request-user')}</button>
        </section>
      </React.Fragment>
    )
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    requestLoggedUser: () => dispatch(GET_LOGGED_USER()),
  }
}

const dashboardWithTranslations = withNamespaces()(injectSheet(styles)(DashboardView))
export default connect(
  null,
  mapDispatchToProps
)(dashboardWithTranslations)
