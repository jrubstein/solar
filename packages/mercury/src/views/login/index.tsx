import React from 'react'
import { Helmet } from 'react-helmet'
import { Map } from 'immutable'
import { withNamespaces, WithNamespaces } from 'react-i18next'
import { Dispatch } from 'redux'
import { WithStyles, StyleRules } from '@material-ui/core/styles'
import injectSheet from 'react-jss'
import { LoginForm } from './components/LoginForm'
import { connect } from 'react-redux'
import { LoginData, LOGIN } from './Actions'

const styles: StyleRules = {
  content: {
    display: 'flex',
    'flex-direction': 'column',
    'justify-content': 'center',
    'text-align': 'center',
  },
  error: {
    color: 'red',
  },
}

interface LoginProps extends WithNamespaces, WithStyles {
  submit: (data: LoginData) => void
  authError?: string
}

class LoginView extends React.Component<LoginProps> {
  constructor(props: LoginProps) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this)
  }

  public render() {
    const { authError, classes, t } = this.props
    return (
      <section className={classes.content}>
        <Helmet>
          <title>{t('login-title')}</title>
        </Helmet>
        <h1>{t('login-title')}</h1>
        {authError && <span className={classes.error}>{authError}</span>}
        <LoginForm onSubmit={this.onSubmit as any} />
      </section>
    )
  }

  private onSubmit(values: Map<string, string>) {
    this.props.submit(values.toObject() as LoginData)
  }
}

const mapStateToProps = ({ login: { authError } }) => {
  return {
    authError,
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    submit: (loginData: LoginData) => dispatch(LOGIN(loginData)),
  }
}

const loginWithTranslations = withNamespaces()(injectSheet(styles)(LoginView))
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(loginWithTranslations)
