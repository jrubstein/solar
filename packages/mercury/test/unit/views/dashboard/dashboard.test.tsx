import renderer from 'react-test-renderer'
import React from 'react'
import { DashboardView } from '../../../../src/views/dashboard'
import { WithNamespaces } from 'react-i18next'
import { i18n } from 'i18next'
import { shallow } from 'enzyme'

const mockI18NProps: WithNamespaces = {
  defaultNS: '',
  i18nOptions: {},
  t: code => code,
  tReady: true,
  i18n: {} as i18n,
}

describe('DashboardView Test', () => {
  const requestLoggedUser = jest.fn()

  beforeEach(() => {
    requestLoggedUser.mockReset()
  })

  describe('render', () => {
    it('should render the dashboard', () => {
      const component = renderer.create(
        <DashboardView requestLoggedUser={requestLoggedUser} {...mockI18NProps} classes={{}} />
      )
      expect(component).toMatchSnapshot()
    })
  })

  describe('requestLoggedUser', () => {
    it('should call requestLoggedUser when the button is clicked', () => {
      const component = shallow(<DashboardView requestLoggedUser={requestLoggedUser} {...mockI18NProps} classes={{}} />)
      component.find('button').simulate('click')
      expect(requestLoggedUser.mock.calls).toHaveLength(1)
    })
  })
})
