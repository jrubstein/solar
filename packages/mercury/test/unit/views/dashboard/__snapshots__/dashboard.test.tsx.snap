// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView Test render should render the dashboard 1`] = `
<section>
  <button
    onClick={[MockFunction]}
  >
    dashbord-request-user
  </button>
</section>
`;